app/scripts/modules/ecs/iamRoles/iamRole.read.service.ts

```diff
@@ -0,0 +1,21 @@
+import { IPromise, module } from 'angular';
+
+import { API_SERVICE, Api } from 'core/api/api.service';
+import { IRoleDescriptor } from './IRole';
+
+export class IamRoleReader {
+  public constructor(private API: Api) { 'ngInject'; }
+
+  public listRoles(provider: string): IPromise<IRoleDescriptor[]> {
+    return this.API
+      .all('roles')
+      .all(provider)
```

Comment:
Does `provider` need to be parameterized? If so, this probably belongs in the core module, not the ECS module.

Reply:
Mmmm @robzienert are roles to be a spinnaker primitive, or not?  If I remember well from a PR comment on Gate, you wanted them to be ECS specific.  What's your recommended approach here?